stablehlo/tests/interpret_probe.mlir

```diff
@@ -0,0 +1,84 @@
+// RUN: stablehlo-translate --interpret --probe-output-dir=%T -split-input-file %s
+
+func.func @probe_i8() {
```

Comment:
Do these particular bidwidth types have special cases? I think testing one bitwidth per element type is fine. Otherwise, consider including tests for all supported element types (incl. 64-bitwidth and complex types).

As far as interpreter workstream goes, we've been testing the highest bitwidth for integer unless different element types are handled differently. Linking the guideline [here](https://github.com/openxla/stablehlo/blob/main/docs/reference.md#testing-guidelines).

This isn't a requirement for this PR, but would be nice to either test highest bitwidth for each element type (bool, int, uint, floats, complex) for element type coverage, or test all bitwidths for all element types. The former is easier to maintain hence why we've been following this for the interpreter while maintaining code coverage.

Reply:
They don't have special cases, moreso that they map to different underlying C++ builtin types for serialization, so I was trying to hit all the different lengths (i.e. a float/int32_t would use the same bitwidth). But I've switched the LIT test to be more inline with the testing guideline, and now am locking down the highest bitwidth for `bool`, `int`, `uint` and `float`.

I also added a test for `complex<f32>` as it can be represented with a `double` in the dynamic type dispatch. However, for `complex<f64>` I think that'd require the `__int128` type on some compilers, so it's `#ifdef`'ed in the dispatch function. Thoughts?

Reply:
Sounds good, thanks.